programs/whirlpool/src/util/v2/remaining_accounts_utils.rs

```diff
@@ -0,0 +1,100 @@
+use anchor_lang::prelude::*;
+use crate::errors::ErrorCode;
+
+#[derive(AnchorSerialize, AnchorDeserialize, Clone, PartialEq, Eq)]
+pub enum AccountsType {
+    TransferHookA,
+    TransferHookB,
+    TransferHookReward,
+    TransferHookInput,
+    TransferHookIntermediate,
+    TransferHookOutput,
+    //TickArray,
+    //TickArrayOne,
+    //TickArrayTwo,
+}
+
+#[derive(AnchorSerialize, AnchorDeserialize, Clone)]
+pub struct RemainingAccountsSlice {
+    pub accounts_type: AccountsType,
+    pub length: u8,
+}
+
+#[derive(AnchorSerialize, AnchorDeserialize, Clone)]
+pub struct RemainingAccountsInfo {
+    pub slices: Vec<RemainingAccountsSlice>,
+}
+
+#[derive(Default)]
+pub struct ParsedRemainingAccounts<'info> {
+    pub transfer_hook_a: Option<Vec<AccountInfo<'info>>>,
+    pub transfer_hook_b: Option<Vec<AccountInfo<'info>>>,
+    pub transfer_hook_reward: Option<Vec<AccountInfo<'info>>>,
+    pub transfer_hook_input: Option<Vec<AccountInfo<'info>>>,
+    pub transfer_hook_intermediate: Option<Vec<AccountInfo<'info>>>,
+    pub transfer_hook_output: Option<Vec<AccountInfo<'info>>>,
+    //pub tick_array: Option<Vec<AccountInfo<'info>>>,
+    //pub tick_array_one: Option<Vec<AccountInfo<'info>>>,
+    //pub tick_array_two: Option<Vec<AccountInfo<'info>>>,
+}
+
+pub fn parse_remaining_accounts<'info>(
+  remaining_accounts: &[AccountInfo<'info>],
+  remaining_accounts_info: &RemainingAccountsInfo,
+  valid_accounts_type_list: &[AccountsType],
```

Comment:
I understand what this function is supposed to do: take a list of remaining_accounts and split them up based on what mint they belong to. But I don't fully understand yet how this function is doing that. Where does `remaining_account_info` come from? Is this something that needs to be specified in the instruction data?

Reply:
``remaining_account_info`` is not an Anchor provided function.
It is a simple data passed as ``data`` of an instruction.

At client-side, we need to build ``remainingAccountsInfo`` data by the following Util class.
https://github.com/orca-so/whirlpools/pull/134/files#diff-1ab45321ec7b9a26165920668ef546656e1fe0d49a6de83c9f0e8aa9352a4cdc

This class is used as follows.
https://github.com/orca-so/whirlpools/pull/134/files#diff-e87787e30f5ee571d972a7cd9b32aa56091ead71895bd8e13fdf0d61c0405f2f

There is flexibility to add a variety of things to remaining_accounts.

However, unless the role of each account is made explicit, we will have to carefully check and categorize account owners, etc. If swap_v2 will be extended to accept additional TickArray accounts, or if other accounts are needed, this data could be used for would make it possible to use remaining_accounts in a structured manner.